server/src/routes/csvFileRoutes.ts

```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+import { convertCSV } from '../controller/csvController'; // Import csv controller
+import { upload } from '../middleware/uploadMiddleware'; // Import the upload middleware
+
+const router = express.Router();
+
+// Define the POST route for CSV conversion, using the upload middleware to handle file uploads
+router.post('/convert-csv', upload.single('file'), convertCSV);
+
+export default router;
```

Comment:
im venturing that the reason your route is not working is because you have not created the route in the main file. also use pre-existing routes.

Reply:
look into how the routes flow from the main entry point to any of the endpoint and you will find the issues.